Implemented discovery registry over metadata

g4s8 opened the pull request:
[#124] - implemented discovery registry over metadata
 - Added new interface `Registry` for discovery package
 - Updated `Discovery` struct to include registry
 - Added implementation of `Registry` over metadata
 - Added a test for meta implementation
 - Removed todo

stepanvalyavskiy reviewed (changes requested):
Hi, @g4s8, I've seen a couple of public methods without comments, please add them.
I will try to enforce this rule via linter.

g4s8 commented:
@stepanvalyavskiy thanks, I've added `golint` with `./discovery` and `./locators` arguments to `Makefile` `lint` step and fixed all warnings, please check new commit

stepanvalyavskiy commented on Makefile:
```diff
@@ -45,6 +44,9 @@ install-deps:
 # run golangci-lint
 lint:
+	golint .
+	golint ./discovery
```
I want to add golint to golangci-lint config. Then it could be configured as other linters and golangci-lint does linters faster then outside one by one.

stepanvalyavskiy replied:
I've done it in #127. @g4s8 please check. Maybe we can merge that PR first, and then revert changes in Makefile here?

stepanvalyavskiy replied:
Or at first merge this PR and then I move golint to golangci-lint config and fix other issues around the project.

g4s8 replied:
@stepanvalyavskiy I've removed these commands from `lint` step

stepanvalyavskiy reviewed (approved):
@g4s8, thanks!
